rust/processor/src/models/token_v2_models/v2_token_activities.rs

```diff
@@ -244,55 +240,24 @@ impl TokenActivityV2 {
                     transaction_timestamp: txn_timestamp,
                 }));
             } else {
-                // This should only happen in the case where we have a burn event and mint event in the same
-                // transaction.
-                if tokens_minted.get(&token_data_id).is_some() {
-                    return Ok(Some(Self {
-                        transaction_version: txn_version,
-                        event_index,
-                        event_account_address,
-                        token_data_id,
-                        property_version_v1: BigDecimal::zero(),
-                        type_: event_type,
-                        from_address: None,
-                        to_address: None,
-                        token_amount: BigDecimal::one(),
-                        before_value: None,
-                        after_value: None,
-                        entry_function_id_str: entry_function_id_str.clone(),
-                        token_standard: TokenStandard::V2.to_string(),
-                        is_fungible_v2: Some(false),
-                        transaction_timestamp: txn_timestamp,
-                    }));
-                }
+                // If the object metadata isn't found in the transaction, then the token was burnt.
+
+                // the new burn event has owner address now!
+                let owner_address = if let V2TokenEvent::Burn(inner) = token_event {
+                    Some(inner.get_previous_owner_address())
+                } else {
+                    // To handle a case with the old burn events, when a token is minted and burnt in the same transaction
+                    None
+                };
 
-                // This should only happen in the case where we have a burn event where the token is gone
-                // and the previous instance of the token wasn't in the batch. We need to look up in the db
-                let latest_nft_ownership =
-                    match CurrentTokenOwnershipV2Query::get_latest_owned_nft_by_token_data_id(
-                        conn,
-                        &token_data_id,
```

Comment:
wait why don't we need this query anymore for the old events?

Reply:
for the old burn events, we set from_address to None. is that fine?